apps/microsoft-teams/frontend/src/components/config/NotificationsSection/NotificationsSection.tsx

```diff
@@ -55,10 +56,45 @@ const NotificationsSection = (props: Props) => {
 
   const updateNotification = (index: number, editedNotification: Partial<Notification>) => {
     const notificationsPayload = [...notifications];
-    notificationsPayload[index] = { ...notificationsPayload[index], ...editedNotification };
+    const existingNotification = notificationsPayload[index];
+
+    // Update the notification at the specified index
+    const updatedNotification = { ...existingNotification, ...editedNotification };
+    notificationsPayload[index] = updatedNotification;
+
+    // Deduplicate based on content
+    const uniqueNotifications = notificationsPayload.filter(
+      (notification, idx) =>
+        notificationsPayload.findIndex(
+          (n) =>
+            n.channelId === notification.channelId && n.contentTypeId === notification.contentTypeId
+        ) === idx
+    );
+
+    // Check if the updated notification is unique
+    const isUnique = uniqueNotifications.length === notificationsPayload.length;
```

Comment:
I think this is a clever way to do this!! 👌  Do we perhaps not want to update the `notificationPayload` on line 63 before checking the uniqueness? If we just run a quick check before doing that, then maybe we don't have to iterate over the `notificationsPayload` "twice". Something like: 

`  notificationPayload.find((n, idx) => n.channelId === updatedNotification.channelId && n.contentTypeId === updatedNotification.contentTypeId) 
`
and then if that renders something ^ we render the modal? Of course there might be nuance here that I am missing with how we manage the notificationsPayload 🤔

Reply:
That's a great call out! With the way the state management is working on these notifications and the way they get persisted to app installation parameters, the `updatedNotification` NEEDS to be set for this to work as intended from what I gather.

I'm having a little trouble following it still but there is a default:

https://github.com/contentful/apps/blob/master/apps/microsoft-teams/frontend/src/constants/defaultParams.ts#L16-L21

default state in reducer
https://github.com/contentful/apps/blob/master/apps/microsoft-teams/frontend/src/components/config/parameterReducer.ts#L54-L67

dispatch
https://github.com/contentful/apps/blob/master/apps/microsoft-teams/frontend/src/components/config/ConfigPage/ConfigPage.tsx#L11

Essentially a blank one gets inserted and over written even on creation, so the UI jumps with an extra notification panel displaying the duplicate notification without filtering 😰

I'm trying to work through this snippet now 🤔 but it still needed filtering logic to avoid the ui skip:

```typescript
const updateNotification = (index: number, editedNotification: Partial<Notification>) => {
  const notificationsPayload = [...notifications];
  const existingNotification = notificationsPayload[index];

  // Update the notification at the specified index
  const updatedNotification = { ...existingNotification, ...editedNotification };

  // See if configured notification exists
  const foundNotificaiton = notificationsPayload.find(
    (n) =>
      n.channelId === updatedNotification.channelId &&
      n.contentTypeId === updatedNotification.contentTypeId
  );

  // If found, open the modal and set the index for further handling
  if (foundNotificaiton) {
    ModalLauncher.open(({ isShown, onClose }) => {
      return (
        <DuplicateModal
          isShown={isShown}
          handleCancel={() => {
            onClose(true);
          }}
          handleConfirm={() => {
            setNotificationIndexToEdit(index);
            onClose(true);
          }}
        />
      );
    });
  }

  notificationsPayload[index] = updatedNotification;

  dispatch({
    type: actions.UPDATE_NOTIFICATIONS,
    payload: notificationsPayload.filter(
      (notifcation) =>
        notifcation.channelId === foundNotificaiton?.channelId &&
        notifcation.contentTypeId === foundNotificaiton.contentTypeId
    ),
  });
};
```

Reply:
ah okay interesting! I think the Set approach is honestly awesome - super easy to understand and very efficient!! 🥇